src/components/Footer/FooterFilter.tsx

```diff
@@ -0,0 +1,58 @@
+import { FC, useState } from 'react';
+import { FilterLink } from '../../types/types';
+import classNames from 'classnames';
+
+interface IProps {
+  items: FilterLink[];
+  getAllTodos: () => void;
+  getActiveTodos: () => void;
+  getCompletedTodos: () => void;
+}
+
+export const FooterFilter: FC<IProps> = ({
+  items,
+  getAllTodos = () => {},
+  getActiveTodos = () => {},
+  getCompletedTodos = () => {},
+}) => {
+  const [selectedItem, setSelectedItem] = useState<string>('All');
+
+  const handleFilter = (filter: string) => {
+    switch (filter) {
+      case 'All':
+        getAllTodos();
+        setSelectedItem(filter);
+        break;
+      case 'Active':
+        getActiveTodos();
+        setSelectedItem(filter);
+        break;
+      case 'Completed':
+        getCompletedTodos();
+        setSelectedItem(filter);
+        break;
+      default:
+        break;
+    }
+  };
```

Comment:
looks too complex, just move selectedItem state to the App.tsx, and pass to this component the selectedItem and setSelectedItem props


after that we can remove this logic
<img width="353" alt="image" src="https://github.com/mate-academy/react_todo-app/assets/70220667/b7c01bca-dba2-4845-9501-ca98514d2f1e">